xdp/src/netlink.rs

```diff
@@ -371,20 +480,22 @@ pub fn netlink_get_neighbors(
             continue;
         }
 
-        let Some(neighbor) = parse_rtm_newneigh(msg, if_index) else {
+        let Some(neighbor) = parse_rtm_newneigh(&msg, if_index) else {
             continue;
         };
 
-        neighbors.push(neighbor);
+        if neighbor.is_valid() {
```

Comment:
this filtering should probably be done in the caller? netlink_get_neighbors
should return all neighbors, not all neighbors that match a hidden impl specific
filter (again, I want this to be generic code anyone can use)

Reply:
I'm starting to get the feeling that all we want here is to filter out `RTM_F_CLONED` entries?